Framework/CurveFitting/src/Algorithms/Fit.cpp

```diff
@@ -48,6 +48,9 @@ void Fit::initConcrete() {
       ->setDocumentation("Whether the fit was successful");
   declareProperty("OutputChi2overDoF", 0.0, "Returns the goodness of the fit",
                   Kernel::Direction::Output);
+  declareProperty("OutputNIterations", 0, Kernel::Direction::Output);
```

Comment:
You can see the breakage in the Python algorithms you had to modify to accommodate the changes in Fit: if a script expects the covariance matrix to be at index 3 in the tuple returned by Fit, the addition of OutputNIterations moves the index to 4. This breaks every script relaying on these indices.

I understand the trick done with the output workspaces, but I don't know if there is a nice way around this. How big of a deal is this really? Maybe someone at @mantidproject/gatekeepers has an opinion on this?